timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceIntegrationTest.java

```diff
@@ -67,79 +76,105 @@
-        LockImmutableTimestampResponse response1 = cluster.timelockService()
+        LockImmutableTimestampResponse response1 = namespace.timelockService()
                 .lockImmutableTimestamp();
-        LockImmutableTimestampResponse response2 = cluster.timelockService()
+        LockImmutableTimestampResponse response2 = namespace.timelockService()
                 .lockImmutableTimestamp();
 
-        long immutableTs = cluster.timelockService().getImmutableTimestamp();
+        long immutableTs = namespace.timelockService().getImmutableTimestamp();
         assertThat(immutableTs).isEqualTo(response1.getImmutableTimestamp());
 
-        cluster.unlock(response1.getLock());
+        namespace.unlock(response1.getLock());
 
         assertThat(immutableTs).isEqualTo(response2.getImmutableTimestamp());
 
-        cluster.unlock(response2.getLock());
+        namespace.unlock(response2.getLock());
     }
 
     @Test
     public void batchedStartTransactionCallShouldLockImmutableTimestamp() {
+        // requestor id corresponds to an instance of the timelock service client
         StartTransactionRequestV4 request = StartTransactionRequestV4.createForRequestor(
                 UUID.randomUUID(),
                 123);
 
-        LockImmutableTimestampResponse response1 = cluster.namespacedClient()
+        LockImmutableTimestampResponse response1 = namespace.namespacedTimelockRpcClient()
                 .startTransactions(request)
                 .immutableTimestamp();
 
-        LockImmutableTimestampResponse response2 = cluster.namespacedClient()
+        LockImmutableTimestampResponse response2 = namespace.namespacedTimelockRpcClient()
                 .startTransactions(request)
                 .immutableTimestamp();
 
-        long immutableTs = cluster.timelockService().getImmutableTimestamp();
+        // above are two *separate* batches
+
+        long immutableTs = namespace.timelockService().getImmutableTimestamp();
         assertThat(immutableTs).isEqualTo(response1.getImmutableTimestamp());
 
-        cluster.namespacedClient().unlock(ImmutableSet.of(response1.getLock()));
+        namespace.namespacedTimelockRpcClient().unlock(ImmutableSet.of(response1.getLock()));
 
         assertThat(immutableTs).isEqualTo(response2.getImmutableTimestamp());
 
-        cluster.namespacedClient().unlock(ImmutableSet.of(response2.getLock()));
-
+        namespace.namespacedTimelockRpcClient().unlock(ImmutableSet.of(response2.getLock()));
     }
 
     @Test
     public void immutableTimestampIsGreaterThanFreshTimestampWhenNotLocked() {
-        long freshTs = cluster.getFreshTimestamp();
-        long immutableTs = cluster.timelockService().getImmutableTimestamp();
+        String randomNamespace = randomNamespace();
+        long freshTs = cluster.client(randomNamespace).getFreshTimestamp();
+        long immutableTs = cluster.client(randomNamespace).timelockService().getImmutableTimestamp();
```

Comment:
I think we can just use the `namespace` constant here? I'd imagine we'd use that for anything where we aren't trying to test specific interactions across namespaces. Unless are we concerned about other tests doing shady things with the immutable timestamp

Reply:
I was getting test failures because we couldn't acquire the lock. I'm thinking to actually make `namespace` be a randomNamespace at the beginning, then we don't have to worry about sharing stuff.

Reply:
OK, yeah so looks like the other tests _do_ do shady things

Reply:
changed so that each test has a random namespace to do stuff